common/persistence/cassandra/history_store.go

```diff
@@ -180,9 +181,16 @@ func (h *HistoryStore) ReadHistoryBranch(
 	} else {
 		queryString = v2templateReadHistoryNode
 	}
-	query := h.Session.Query(queryString, treeID, branchID, request.MinNodeID, request.MaxNodeID)
+
+	var orderSuffix string
+	if request.ReverseOrder {
+		orderSuffix = v2templateReadHistoryNodeReverseSuffix
+	}
+
+	query := h.Session.Query(queryString+orderSuffix, treeID, branchID, request.MinNodeID, request.MaxNodeID)
```

Comment:
instead of building the query for each request, how about make the query string a const? e.g. v2templateReadHistoryNodeMetadata
v2templateReadHistoryNode
v2templateReadHistoryNodeReverse

Reply:
MetadataOnly can be changed to Mode or RequestType or ResultType